editor/plugins/mesh_instance_3d_editor_plugin.cpp

```diff
@@ -301,47 +301,81 @@ void MeshInstance3DEditor::_menu_option(int p_option) {
 			ur->commit_action();
 		} break;
 		case MENU_OPTION_CREATE_UV2: {
-			Ref<PrimitiveMesh> primitive_mesh = Object::cast_to<PrimitiveMesh>(*node->get_mesh());
+			Ref<Mesh> mesh2 = node->get_mesh();
+			if (!mesh.is_valid()) {
+				err_dialog->set_text(TTR("No mesh to unwrap."));
+				err_dialog->popup_centered();
+				return;
+			}
+
+			// Test if we are allowed to unwrap this mesh resource.
+			String path = mesh2->get_path();
+			int srpos = path.find("::");
+			if (srpos != -1) {
+				String base = path.substr(0, srpos);
+				if (ResourceLoader::get_resource_type(base) == "PackedScene") {
+					if (!get_tree()->get_edited_scene_root() || get_tree()->get_edited_scene_root()->get_scene_file_path() != base) {
+						err_dialog->set_text(TTR("Mesh cannot unwrap UVs because it does not belong to the edited scene. Make it unique first."));
+						err_dialog->popup_centered();
+						return;
+					}
+				} else {
+					if (FileAccess::exists(path + ".import")) {
+						err_dialog->set_text(TTR("Mesh cannot unwrap UVs because it belongs to another resource which was imported from another file type. Make it unique first."));
+						err_dialog->popup_centered();
+						return;
+					}
+				}
+			} else {
+				if (FileAccess::exists(path + ".import")) {
+					err_dialog->set_text(TTR("Mesh cannot unwrap UVs because it was imported from another file type. Make it unique first."));
+					err_dialog->popup_centered();
+					return;
+				}
+			}
+
+			Ref<PrimitiveMesh> primitive_mesh = Object::cast_to<PrimitiveMesh>(*mesh2);
```

Comment:
I think this can just be this:
```suggestion
			Ref<PrimitiveMesh> primitive_mesh = mesh2;
```
The ref will only be valid if the cast is valid.